### PR TITLE
Fix gardenlet rbac for events in seed cluster

### LINKED ISSUE #4625: Gardenlet is missing read permissions for events in the seed cluster
**How to categorize this issue?**

/area ops-productivity performance quality
/kind bug
/priority 3

**What happened**:
Gardenlet was failing for quite some time in newly created seed cluster with error like 
```
{"level":"info","msg":"Waiting until the nginx-ingress-controller service is ready...","seed":"foo","time":"2021-09-08T11:55:13Z"}
{"level":"error","msg":"error retry failed with context deadline exceeded, last error: nginx-ingress-controller service is not ready: `.status.loadBalancer.ingress[]` has no elements yet, i.e. external load balancer has not been created occurred while waiting for load balancer to be ready","seed":"foo","time":"2021-09-08T11:55:18Z"}
E0908 11:55:18.308357       1 reflector.go:138] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:241: Failed to watch *v1.Event: failed to list *v1.Event: events is forbidden: User "system:serviceaccount:garden:gardenlet" cannot list resource "events" in API group "" at the cluster scope
```

While it has no controller actively watching `events`, during the bootstrapping of the seed, if the managed ingress is not disabled, a LB will be created and waited its public IP to be provisioned here https://github.com/gardener/gardener/blob/ec9152a73f63642f40907ed079bd2d0c6ff732a3/pkg/operation/seed/seed.go#L1000 In case of failure, `events` will be looked for errors here https://github.com/gardener/gardener/blob/ec9152a73f63642f40907ed079bd2d0c6ff732a3/pkg/utils/kubernetes/kubernetes.go#L200. But gardenlet is not allowed to read events, it has only create, patch and update permissions in the `garden` namespace currently https://github.com/gardener/gardener/blob/ec9152a73f63642f40907ed079bd2d0c6ff732a3/charts/gardener/gardenlet/charts/runtime/templates/role-garden-gardenlet.yaml#L15-L22.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Deploy a fresh installation of gardenlet with managed ingress enabled into a cluster that cannot successfully provision new LB services (e.g. scaled down CCM or restricted quota).

**Anything else we need to know?**:
Thanks to @timebertt for the findings!

**Environment**:

- Gardener version: v1.30.1
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area security
/kind bug

**What this PR does / why we need it**:

Fix gardenlet rbac for events in seed cluster (see commits / issue for more details)

**Which issue(s) this PR fixes**:
Fixes #4625

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/client/kubernetes/clientmap/internal/seed_clientmap.go [hunk 1]
```diff
@@ -65,6 +67,10 @@ func (f *SeedClientSetFactory) NewClientSet(ctx context.Context, k clientmap.Cli
 				Scheme: kubernetes.SeedScheme,
 			},
 		),
+		kubernetes.WithUncached(
+			&corev1.Event{},
+			&eventsv1.Event{},
+		),
```
Comment (stoyanr) [inline pkg/client/kubernetes/clientmap/internal/seed_clientmap.go:73]: I think the unit tests should be adapted to this change.

Comment (timebertt) [reply pkg/client/kubernetes/clientmap/internal/seed_clientmap.go:73]: Of course, done.

**Review by stoyanr — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**